mesonbuild/rewriter.py

```diff
@@ -1077,10 +1157,21 @@ def run(options: argparse.Namespace) -> int:
         if not isinstance(commands, list):
             raise TypeError('Command is not a list')
 
-        for i in commands:
-            if not isinstance(i, object):
+        for i, cmd in enumerate(commands):
+            if not isinstance(cmd, object):
                 raise TypeError('Command is not an object')
-            rewriter.process(i)
+            rewriter.process(cmd)
+
+            if i == len(commands) - 1: # Improves the performance, is not necessary for correctness.
+                continue
+
+            rewriter.apply_changes()
```

Comment:
You could move this line before testing for last command, and remove the call to apply_changes after the loop (unless you need to apply changes even if there are no commands...)

Reply:
Fixed